testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ejb/DefaultElytronEjbSecurityDomainTestCase.java

```diff
@@ -0,0 +1,149 @@
+                .addPackage(EntryBean.class.getPackage())
+                .addClass(Util.class)
+                .addClass(WhoAmI.class)
+                .addClass(org.jboss.as.controller.operations.common.Util.class)
+                .addPackage(AbstractSecurityDomainSetup.class.getPackage())
+                .addAsResource(currentPackage, "users.properties", "users.properties")
+                .addAsResource(currentPackage, "roles.properties", "roles.properties")
+                .addAsManifestResource(currentPackage, "jboss-ejb3.xml", "jboss-ejb3.xml")
+                .addAsManifestResource(createPermissionsXmlAsset(new ElytronPermission("getSecurityDomain")), "permissions.xml");
+    }
+
+    @Test
+    public void testSecurityOnTwoBeansInAbsenceOfExplicitSecurityDomain() throws Exception {
+
+        final EjbUnsecuredBean unsecuredBean = InitialContext
+                .doLookup("java:module/" + EjbUnsecuredBean.class.getSimpleName());
+
+        final String echoResult = unsecuredBean.echo("unsecuredBeanEcho");
+        assertEquals("unsecuredBeanEcho", echoResult);
+
+        final Callable<Void> callable = () -> {
+            try {
+                final Principal principal = securedBean.getCallerPrincipal();
+                assertNotNull("EJB must never return a null from the getCallerPrincipal method.", principal);
+                assertEquals("user1", principal.getName());
+            } catch (RuntimeException e) {
+                fail("EJB must provide the caller’s security context information during the execution of a business method (" + e.getMessage() + ")");
+            }
+            return null;
+        };
+        Util.switchIdentitySCF("user1", "password1", callable);
+    }
+
+    public static class DefaultElytronEjbSecurityDomainTestCaseSecuritySetup extends AbstractSecurityDomainSetup {
+
+        protected static final String DEFAULT_SECURITY_DOMAIN_NAME = "elytron-tests";
+        private ElytronDomainSetup elytronDomainSetup;
+        private EjbElytronDomainSetup ejbElytronDomainSetup;
+        private ServletElytronDomainSetup servletElytronDomainSetup;
+
+        protected String getUsersFile() {
+            return new File(DefaultElytronEjbSecurityDomainTestCase.class.getResource("users.properties").getFile()).getAbsolutePath();
+        }
+
+        protected String getGroupsFile() {
+            return new File(DefaultElytronEjbSecurityDomainTestCase.class.getResource("roles.properties").getFile()).getAbsolutePath();
+        }
+
+        @Override
+        protected String getSecurityDomainName() {
+            return DEFAULT_SECURITY_DOMAIN_NAME;
+        }
+
+        @Override
+        public void setup(final ManagementClient managementClient, final String containerId) throws Exception {
+
+            // elytron profile is enabled
+            elytronDomainSetup = new ElytronDomainSetup(getUsersFile(), getGroupsFile(), getSecurityDomainName());
```

Comment:
These three setup tasks never have their tearDown methods called, so the changes they make will not be reverted.